include/revng/PipeboxCommon/Helpers/PipeRunPipes/LLVMFunctionPipeRun.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/IR/Function.h"
+#include "llvm/Support/Progress.h"
+
+#include "revng/Model/IRHelpers.h"
+#include "revng/PipeboxCommon/Helpers/PipeRunPipes/Base.h"
+#include "revng/PipeboxCommon/Helpers/PipeRunPipes/Helpers.h"
+#include "revng/PipeboxCommon/LLVMContainer.h"
+#include "revng/Support/Tag.h"
+
+namespace detail {
+
+template<size_t Index, typename... ArgsT>
+revng::pypeline::LLVMFunctionContainer &getLLVMContainer(ArgsT &...Args) {
+  std::tuple<ArgsT &...> Tuple(Args...);
+  return std::get<Index>(Tuple);
+}
+
+} // namespace detail
+
+template<typename T>
+concept IsLLVMFunctionPipeRun = requires(T &PipeRun,
+                                         const model::Function &Function,
+                                         llvm::Function &LLVMFunction) {
+  requires IsBasePipeRun<T>;
+  { PipeRun.runOnFunction(Function, LLVMFunction) } -> std::same_as<void>;
+};
+
+// This class wraps a PipeRun that implements a `runOnFunction` with both the
+// `model::Function` and `llvm::Function`. Since there a few member objects
+// that are exposed by the PipeRun (e.g. `Arguments`) and a few optional
+// methods that might be exposed (e.g. `checkPrerequisites`) this class
+// inherits the PipeRun so that it doesn't have to "proxy" everything, which
+// would be required if the class was encapsulated into this one.
+template<IsLLVMFunctionPipeRun T>
+class LLVMFunctionPipeRun : public T {
```

Comment:
Rotate inheritance. Turn `LLVMFunctionPipeRun` into a mixin/CRTP so that what's currently `T` becomes actually a pipe run and we no longer have a "semi-pipe run".